Fixes #6206: Fixes the enlarging of text in RTE

bansalnitish opened the pull request:
Fixes #6206.

bansalnitish committed:
Fixes enlargement of text

anubhavsinha98 committed:
Fix #6004: Added a presubmit check to ensure imports of JS files are sorted (#6125)

* First

* First

* Second

* Third

* Fourth

* Fixed imports order

* Few more

* 1800

* 1900

* Removed

* corrected

* 2010

* Final

* resolved linting issues

* resolved linting issues

* Fixed HintsAndSolution

* Fixed Hintandsolution

* Fixed ExplorationCreationService.js

* Added presubmit check

* Fixed all files

* Added one time parsing only

* Made some changes

* Update pre_commit_linter.py

* changes done-linitng issues fixed

* Fixed the changes

* Updated the changes

* Changes done

* Fixed linting

* Fixed failed travis ci check

* Fixed changes

* Fixed changes

* Fixed linting issues

* Changes done

* Updated LocalNav.js

ujjwalgulecha committed:
Fix #4057 Make PlayerPositionService.js 100% test coverage (#6258)

* Fix #4057 Make PlayerPositionService.js 100% test coverage

* address comments

* minor refactor

* minor refactor

seanlip commented on core/templates/dev/head/css/oppia.css:
```diff
@@ -352,6 +352,12 @@ p:last-child {
   margin: 0px;
 }
 
+/* This will prevent the enlargment of the text that is nested inside the p
```
This is too broad. It affects all `p span` tags, not just RTE ones. In addition using !important typically indicates a problem with the approach. 

Can you use a more specific CSS rule that is only scoped to the RTE viewer component? That may make it possible for you to do this without !important.

bansalnitish replied:
I think, it would affect a lot of things but it won't break any of them. The nested em size units add up as we go deeper. The `font-size: 1em`, removes this addition and leads to inherit the size from parent itself. I don't think we need a different size in span tag (which is different from that in p tag). Does this sounds good?

There is inline css which is being introduced with the span tag. Can we override that css by making it more specific without using !important? I guess, no. I referred this for the same: https://stackoverflow.com/questions/16813220/how-can-i-override-inline-styles-with-external-css/16813263.

Thanks!

seanlip replied:
Hm, I think this means we really do need to figure out where the inline CSS is coming from. I added a comment on the original issue thread. Thanks for the explanation!

seanlip reviewed (changes requested):
Hi @bansalnitish, thanks, took a pass! One question, though -- could you please also explain, in a top-level comment, why the error is occurring? 

EDIT: ah sorry, missed your comment in the issue itself. Will reply there.

nithusha21 commented:
Hi @bansalnitish, could you please add a changelog label (before merging) as mentioned [here](https://groups.google.com/forum/#!topic/oppia-dev/4p555mIskWE). Thanks!

BenHenning committed:
Correct comment string in UserSubscriptionsModel (#6310)

The original string seemed backwards based on how this is used (see https://github.com/oppia/oppia/blob/0d676f9c73d16639b05273e61be683e19fd604e2/core/domain/subscription_services.py#L145). It appears that this field, as expected, is actually storing the IDs of creators this user is subscribed to.

nithusha21 committed:
Add issue template for server errors (#6308)

* add issue template for server errors

* improve about section

* review changes

* minor nits

seanlip committed:
Add a script for updating indexes. Move gcloud functions into a separate module. (#6305)

* Add a script for updating indexes. Move gcloud functions into a separate file.

* Further changes needed to use gcloud for deployment.

* Remove unused code.

* Add ability to specify custom versions.

* Do not switch to new version after uploading.

* Update fileoverview docstring.

bansalnitish committed:
Fix #2306: Fixes signup issue when a new tab is opened (#6235)

* Fix signup issue

* Fix failing test

* modal to signup and added test

* removed useless variable

* Add docstrings and modify function names

* Addressed review comments

* Replace post with post_json

* Addressed review comments

* Fixed lint errors

* Addressed review comments

apb7 committed:
Port back-end tests from Travis to CircleCI (#6234)

* Basic config

* Minor nit

* Test caching mechanism

* Try integration of frontend and backend tests

* Syntax fix

* Add defaults back to each job

* Add checkout for each job

* Add checkout to setup job

* Add anchor for restore_cache

* Add e2e tests accessiblity

* Add third_party_cache and setup_cache key

* Fix cache paths

* Change working dir path

* Use abs paths for cache

* Try specifying chrome version

* Add libappindicator

* Tweak

* Try fixing path and run only e2e tests

* Add lint, frontend and backend tests to config.yml

* Remove lint, frontend and backend tests from travis.yml

* Remove undefined anchor

* Fix yml

* Install dependencies

* Minor tweaks

* Move init hook to Master section in pylintrc

* Test commit: try changing the working dir

* Port backend tests from Travis to CircleCi

* Revert changes made in pylintrc

* Revert changes to Travis and comment out backend and frontend tests' run

* Run frontend and generate coverage

* Install codecov as a dependency

* Add sudo to overcome permission error

* Re-order codecov install

* Add branches to be build and try enabling backend test coverage

* Fix config

* Remove branch section

* Try: lint tests

* Try: lint tests

* Try: Add dependency installation to lint tests job

* Try: Caching

* Try: Caching

* Try: Caching

* Try: Caching

* Try: Caching

* Try: Caching

* Run lint, frontend and backend tests

* Generate date for each job

* Comment out lint test and re-order third party installation

* Fix Travis

* Remove --generate_coverage_report flag from backend tests job

* Add explanation to travis config for commenting checks out

* Re-order dependency install

* Re-order dependency

* Revert: Remove --generate_coverage_report flag from backend tests job

* Remove third party install from lint check

ankita240796 committed:
Fix part of #6254: Refactor for directives.js and filters.js (#6275)

* Refactor for directives.js

* Remove unused directives

* Refactor for filters.js

* Fix issues

* Add constants

* Move scripts to where they are required

* Add removed service

* Fix failing test

* Fix failing test

apb7 committed:
Run lint tests on Travis temp (#6328)

apb7 commented:
Hi @bansalnitish, please bring your PR branch up-to-date with oppia/develop. Thanks!

seanlip committed:
Update mailing list for Travis emails. (#6330)

DubeySandeep committed:
Added more codeowners realted to frontend pages. (#6332)

DubeySandeep committed:
Remove google forum link. (#6333)

* Removed google forum.

* Removed forum from frontend.

* Added review changes.

aks681 committed:
Fix #6210: Added an exploration whitelist (#6255)

* fix #6210

* fixed issue

* fix lint

* fixed merge conflicts

* minor change

* made review changes

* made review changes

* made review change

import-keshav committed:
Fix #6214: Audio progress bar hides for auto-generated audio (#6265)

mzaman07 committed:
Fixes #5784: unhandled promise in the ng-audio library. (#6280)

DubeySandeep committed:
Adds written translations domain object in frontend for state and question. (#6320)

* Added written translations in frontend.

* Added review changes and fixed failing tests.

* Imported new object file in exploration player.

* Added review changes.

* Added newline after each test case.

anubhavsinha98 committed:
Added subtopic_viewer (SubtopicPageDataHandler), which returns page contents for a subtopic. (#6327)

* First

* First

* Second

* Third

* Fourth

* Added subtopic_viewer.py

* Few changes

* Added the sevices

* Added new test file

* Made Some Changes

* Done

* Fixed linting checks

* Fixed linting checks

* Completed

* To Restart Tests

* base_test run success

* subtopic_id int

* Type casted

DubeySandeep committed:
Adds QA team as a codeowner for test files. (#6352)

* Added QA team to e2e test files.

* Added review changes.

DubeySandeep committed:
Generalized landing page based on subject and topic. (#6357)

amey-kudari committed:
Fix #6301: fixes Dev-mode label showing up for a brief amount of time on page load.  (#6356)

* added ng cloak to fix dev_mode showing up

* modify DIV MODE def in constants.js

* added the ng-cloak update in base_test.py

* linting errors

* lint errors

* removed css, and replicated changed in tests

DubeySandeep committed:
Moved state related components in a separate dir. (#6334)

* Moved state related components in a seperate dir.

* Added proper url for html files.

import-keshav committed:
Fixes #6316: Refactor base.html (#6319)

* 1st commit (remove scripts of creator dashboard and topics and skill dashboard)

* remove filter ConvertToPlainTextFilter from base.html

* undo commit remove scripts of creator dashboard and topics and skill dashboard

* remove truncate filter from base.html

* remove normalizeWhitespace filter from base.html

* remove ValidatorsService script import from base.html

* fixes lints issue

* not removed script of ValidationService

* fies tests issue related to import of convertToPlainText filter

fixes e2e tests

* fix test due to error on profile.html

* fixes e2e tests issue of ConvertToPlainTextFilter importing

* fix changes

DubeySandeep committed:
Replace google-map with openStreetMap in map interaction. (#6309)

* Replaced google-map with openStreetMap in interaction.

* Changes made after testing UI on mobile devices.

* Simple UI changes.

* Updated correct depen. URL.

* Fixed state navigation issue.

* Fixed lint issue.

* Added review changes.

* Added review changes.

* Made some stylistic changes after following a user journey.

* Remove googlemaps from node_modules.

* Added leaflet type for npm.

* made changes in tsconfig.json

* Added CSS class.

abhaygupta97 committed:
Fix #2584: Replace default text while inserting links with placeholder (#6286)

* Fixes #2584: Replace default text while inserting links with placeholder

* Making minor changes

* Modifying backend test

* Fixing e2e test errors

* Making changes

* Creating new html file for sanitized url

* Adding validator to SanitizeUrl object

* Adding spaces

* Adding tests

* Adding else statement

* Replacing list_editor with sanitized_url_editor

seanlip committed:
Fix frontend tests failing occasionally. (#6371)

aks681 committed:
New Structures e2e tests (#6315)

* added topics and skills dashboard e2e tests

* fix lint

* fixed test

* added topic and skill deletion tests

* Created test files

* Modified conf files

* added topic name, description and create story tests

* added story editor tests

* Add E2E tests for Skill Editor

* changes

* Added skill publishing tests

* Lint

* added subtopic tests

* Fix #5692 and done with all e2e tests

import-keshav committed:
Fixes #6226: Moves generating CKEditor widgets from app.js to separate file.  (#6288)

* fixes merge conflicts

* fixes docs issue

* fix docs changes

* rename CkEditorWidgetsGenerator to CkEditorWidgetsInitializer

* fixes tests

* fix changes

* fix merge mistakes

EricZLou committed:
Fix part of #5134: learner_playlist_services - fix grammar and whitespace (#6291)

* Fixes part of #5134: learner_playlist_services

100% coverage
Fixed 2 grammar mistakes and removed trailing whitespaces.

* Update learner_playlist_services_test.py

* Fix test naming and linting errors

* Fixed test naming and linting errors

* Fixed test naming and linting errors

* Fixed test naming and linting errors

* Fixed test naming and linting errors

* took away pos / no pos

* took away pos / no pos

* took away pos / no pos

* took away pos / no pos

* Made tests more descriptive

* Made tests more descriptive

* Added more assert statements

* Added more assert statements

* Added more assert statements

* Added more assert statements

* 100% coverage for learner_playlist_services

* 100% coverage for learner_playlist_services

aks681 committed:
Fixed issues from intuitiveness testing on New Structures (#6314)

* Fixed issues

* fixed more issues

* Made review changes

* Lint

* Made review changes

* Fix #6353

* Made review change

* Added validation for content_ids.

* made review changes

BenHenning committed:
Update release script to use SSH Git URL instead of HTTPS. (#6382)

abhaygupta97 committed:
Fixes #6199: Hint tab failed to open when active hint deleted issue fixed (#6243)

jameesjohn committed:
Fix part of #5002: Moved SITE_FEEDBACK_FORM_URL from GLOBALS to constants.js. (#6365)

* moved SITE_FEEDBACK_FORM_URL to constants.js.

* fix failing circle ci test

* placed constant imports in sorted order

* removed extra new line character at the end of Base.js and deleted unnecessary npm-debug.log

* removed SITE_FEEDBACK_FORM_URL from remaining locations

* moved SITE_FEEDBACK_FORM_URL from necessary files in core/tests/build_sources

* removed constant SITE_FEEDBACK_FORM_URL from base.py

* replaced SITE_FEEDBACK_FORM_URL with DEV_MODE

* removed unnecessary text from jinja_escaping.html

* added dummy url as SITE_FEEDBACK_FORM_URL

brianrodri committed:
update ownership (#6392)

DubeySandeep committed:
Adds written translation property in skillContents and subtopic. (#6241)

* Adds TranslationScript domain object with tests.

* Added ContentTranscriptions object in state and added test for the same.

* Fixed some style issues.

* Added more validation check.

* Fixed some lint issues.

* Added functionality to internally manage adding and deleting content ids for assets dict.

* Trying fixing test errors.

* Fixed test issues.

* Fixed backend test issue.

* Added migration in state and exploration.

* Added new methods as for requirments.

* Updated with translation-script-m1 branch and resolved question tests issues.

* Added review changes.

* Added review changes.

* Some writing changes.

* Spell fix.

* Removed StateContentIdsToAudioTranslationsService from exploration editor tab.

* Relevent changes made in question structure.

* Fixed few issues raised while resolving merge conflicts.

* Added more tests.

* Added review changes.

* Resolved test issues.

* Added written translations property in new structure project.

* Fixed test issues.

* Added comment for constant in feconf.

* Added review changes.

* Update with upstream/develop.

* Fixed test issues.

aks681 committed:
Added pending New Structures e2e tests (#6385)

* Added subtopic page tests

* added question creation tests

siddhantkhandelwal committed:
Fix part of #5134: Increased Coverage from 89% to 100% for core.domain.email_manager.py (#6306)

* Fixes part of #5134: Increased Coverage from 89% to 100% for core.domain.email_manager.py

* Fixes part of #5134: Increased Coverage from 89% to 100% for core.domain.email_manager.py

* Fix part of #5134: Increased Coverage from 89% to 100% for core.domain.email_manager.py

ankita240796 committed:
Fix part of #6343: Update js files to remove typescript compile errors (#6344)

* Fix type issues

* fix more errors

* Fix subsequent declaration errors

* Add global third party typings

* Remove assets from migration

* Fix more errors

* Revert changes for test files

* Address review comments

* Add type definition for visualization

* Fix test

* fix left out error

* Address review comments

* Fix left out issue

* Fix braces

* Fix unicode errors

YashJipkate committed:
Fix faulty layout of exploration graph (#6402)

DubeySandeep committed:
removed unwanted validation checks and related tests. (#6407)

abhaygupta97 committed:
Fix #6370: Fix Title and Goal label alignment (#6372)

DubeySandeep committed:
Added sean to manifest.json (#6421)

seanlip committed:
Make new model IDs only contain alphanumeric characters. (#6408)

* Make new model IDs only contain alphanumeric characters.

* Add test for alphanumeric chars.

jacoblipech committed:
Fix part of #5134: Make test coverage of core.controllers.collection_view 100% (#6427)

Fix part of #5134: Make test coverage of core.controllers.collection_viewer 100%

ankita240796 committed:
Fix part of #6343: Adds custom type definitions (#6430)

* Update typings

* Remove unused vars, make naming consistent

vinitamurthi committed:
Fix #6377: Create Practice Sessions Page Skeleton (#6381)

* Create Practice Sessions Page Skeleton

* Deindent brace by 4 spaces

abhaygupta97 committed:
Fixes part of #5002: Remove logoutUrl from HTML (#6401)

* Fixes part of #5002: Remove logoutUrl from HTML

* Adding urls to profile.py

* Fixing mistakes

* Making suggested changes

* Remove logout url

* Making changes

* Making requested changes

* Removing str from Logout page

anubhavsinha98 committed:
Fix #6100 : Added verbose and non verbose mode in the pre submit check (#6146)

* First

* First

* Second

* Third

* Fourth

* Added Mode

* Added verbose and non verbose mode

* Default non verbose mode

* Changed the arg mode to flag

* Changed the arg mode to flag

* Added -v flag

* removed print statement

* Update pre_commit_linter.py

* Update pre_commit_linter.py

* Changes done-global added

* Fixed changes

* Changes done

Added space after #

* Removed spaces

* Added class

* Added the LintsCheckManager

* Added functions as methods

* Moved to class

* Completed and Fixed Linting issues

* Added comments

* Changes Done

* Bool var name changed and few changes done

* Changes Done

* Changes Done

* changes done

* Made the changes

* Changes Done

* Changes Done

* Done the changes

* Removed

* Added verbose flag to make Travis build unstale

* Added 4 spaces

* Added filenames

* Added filepath

* added filepath

mzaman07 committed:
Fix #6248 : Prevent guest and unauthorized users from drag-and-drop audio in translation tab. (#6304)

amey-kudari committed:
Fix #6325 : modify style of active and inactive submenus on learner dashboard (#6358)

* added background and a subtle margin to selected li

* added background and a subtle margin to selected li

* modify color to hexa value, select color: blue

anubhavsinha98 committed:
Fixes #6205 : Disabled the reject button, enabled only when some review is entered. (#6384)

* First

* First

* Second

* Third

* Fourth

* Bug fixed

* Tooltip

* Added the message

import-keshav committed:
Fixes  #6316: Refactor base.html (#6399)

* remove topics,skill and story creation service from base.html

* remove ConstructTranslationIdsService from base.html

* remove StateRulesStatsService

seanlip committed:
Reorder CODEOWNERS to ensure that the most important rules are at the end. (#6438)

DubeySandeep committed:
Added filter filer in exploration_player. (#6437)

bansalnitish committed:
Merge branch 'fixLargeText' of github.com:bansalnitish/oppia into fixLargeText

bansalnitish committed:
Narrower CSS

bansalnitish commented:
I'm closing this PR. I'll open another one.
